bindings/rust/src/rows.rs

```diff
@@ -87,9 +87,12 @@ unsafe impl Send for Row {}
 unsafe impl Sync for Row {}
 
 impl Row {
-    pub fn get_value(&self, index: usize) -> Result<Value> {
-        let value = &self.values[index];
-        match value {
+    pub fn get_value(&self, idx: usize) -> Result<Value> {
+        let val = &self
+            .values
+            .get(idx)
+            .ok_or(Error::SqlExecutionFailure("invalid row index".to_string()))?;
```

Comment:
could you change this to use `ok_or_else`? otherwise we will be allocating and dropping errors even on every successful call